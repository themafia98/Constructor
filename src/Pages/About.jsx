import React, {Fragment} from 'react';
import {NavLink} from 'react-router-dom';
import Header from '../components/header/Header';


const config = require('../config.json');

class About extends React.Component {

    render(){

        return (
            <Fragment>
                <Header><NavLink to = '/'><h3>{config.title}</h3></NavLink></Header>
                <section className = 'About'>
                    <div className = 'container'>
                        <div className = 'col-12'>
                            <p>Developer: {config.about.developer}</p>
                            <p>Year: {config.about.year}</p>
                        </div>
                    </div>
                </section>
            </Fragment>
        )
    }
}

export default About;