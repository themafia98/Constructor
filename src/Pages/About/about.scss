.information {
    text-align: center;

    p,span {
        font-size: 1.5rem;
    }
    
    span {
        font-weight: bold;
    }

    .logoAbout {
        width: 60%;
    }
}

@media screen and (max-width: 1100px){
    .information{
        .logoAbout {
            width: 70%;
        }
    }


}

@media screen and (max-width: 650px){

    .information{
        .logoAbout {
            width: 100%;
        }
    }
}