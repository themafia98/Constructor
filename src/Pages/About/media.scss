@media screen and (max-width: 1100px){
    .information{
        .logoAbout {
            width: 70%;
        }
    }


}

@media screen and (max-width: 650px){

    .information{
        .logoAbout {
            width: 100%;
        }
    }
}