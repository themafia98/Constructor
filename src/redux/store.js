import {createStore} from 'redux';
import combineReducers from './reducers';

let broswerPluginDev = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const store = createStore(combineReducers,broswerPluginDev);
export default store;