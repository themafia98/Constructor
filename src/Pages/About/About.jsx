import React,{Fragment} from 'react';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';
import {connect} from 'react-redux';


import withFirebase from '../../components/withFirebase';
import Header from '../../components/header/Header';
import Loader from '../../components/loading/Loader';

import './about.scss';

class About extends React.PureComponent {

    static propTypes = {
        config: PropTypes.object.isRequired, /** @Config app */
        firebase: PropTypes.object.isRequired, /** @firebase class for use firebase functions */
        history: PropTypes.object, /** @Router HTML5 history */
        location: PropTypes.object, /** @Router */
        match: PropTypes.object, /** @Router */
    }

    state = {
        config: this.props.config
    };

    render(){

        let user = this.props.firebase.getCurrentUser();
        if (user && this.props.idUser) {
            const { config } = this.state;
            return (
                <Fragment>
                    <Header title = {config.title} idUser = {user.uid} />
                    <section className = 'About'>
                        <div className = 'container'>
                            <div className = 'col-12'>
                                <div className = 'About__information'>
                                <img alt = 'img'
                                    className = 'information__logoAbout' 
                                    src = {process.env.PUBLIC_URL + '/img/about.jpg'}
                                />
                                    <p className ='developer'><span>Developer:</span>
                                                            {config.about.developer}
                                    </p>
                                    <p className = 'year'><span>Year:</span> {config.about.year}</p>
                                </div>
                            </div>
                        </div>
                    </section>
                </Fragment>
            )
        } else if (!user) return <Redirect to = { '/'} />
        else return <Loader path = '/img/loading.gif' type = 'About' />

    }
}

const mapStateToProps = (state) => {
    return {
      idUser: state.cabinet.idUser,
    }
  };

export default connect(mapStateToProps)(withFirebase(About));