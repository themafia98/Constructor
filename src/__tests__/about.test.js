import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';
import About from '../Pages/About/About';
import store from '../redux/testReduxStore';


const firebase = {
    getCurrentUser: () => true
};
const config = require('../config.json');

const AboutComponent = renderer.create(
    <MemoryRouter initialEntries = {['/Cabinet/About']}>
    <Route path = "/Cabinet/About" exact >
        <Provider store = {store}>
            <About config = {config} idUser = {'asdasd'} firebase = {firebase} />
        </Provider>
    </Route>
    </MemoryRouter>
);

test("About test", function(){

    let snapshotAboutComponent = AboutComponent.toJSON();
    expect(snapshotAboutComponent).toMatchSnapshot();
});