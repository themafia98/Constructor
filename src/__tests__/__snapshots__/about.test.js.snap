// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About test 1`] = `
Array [
  <header>
    <div
      className="container"
    >
      <div
        className="flex-row"
      >
        <div
          className="header__CabinetInfo"
          onClick={[Function]}
        >
          <img
            alt="icon"
            className="mainIcon"
            draggable={false}
            onClick={null}
            src="/img/logo.svg"
            title={null}
          />
          <h3
            className="tilteApp"
          >
            Constructor
          </h3>
        </div>
        <div
          className="Navigator"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-w-14  about fas fa-2x fa-book"
            data-icon="book"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <img
            alt="icon"
            className="about"
            draggable={false}
            onClick={[Function]}
            src="/img/about_logo.svg"
            title="about"
          />
          <img
            alt="icon"
            className="logoutButton"
            draggable={false}
            onClick={[Function]}
            src="/img/logOut.svg"
            title="logOut"
          />
        </div>
      </div>
    </div>
  </header>,
  <section
    className="About"
  >
    <div
      className="container"
    >
      <div
        className="col-12"
      >
        <div
          className="About__information"
        >
          <img
            alt="img"
            className="information__logoAbout"
            src="/img/about.jpg"
          />
          <p
            className="developer"
          >
            <span>
              Developer:
            </span>
            Pavel Petrovich
          </p>
          <p
            className="year"
          >
            <span>
              Year:
            </span>
             
            2019
          </p>
        </div>
      </div>
    </div>
  </section>,
]
`;
